aspnet/fundamentals/owin.rst

```diff
@@ -1,8 +1,386 @@
+
+.. image:: owin/_static/websocket-test.png
+
+
+OWIN keys
+---------
+
+OWIN depends heavily on an ``IDictionary<string,object>`` used to communicate information throughout an HTTP Request/Response exchange. ASP.NET 5 implements all of the required and optional keys outlined in the OWIN specification, as well as some of its own. Note that any keys not required in the OWIN specification are optional and may only be used in some scenarios. When working with OWIN keys, it's a good idea to review the list of `OWIN Key Guidelines and Common Keys <http://owin.org/spec/spec/CommonKeys.html>`_
+
+Request Data (OWIN v1.0.0)
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. list-table::
+	:header-rows: 1
+	
+	* - Key
+	  - Value (type)
+	  - Description
+	* - owin.RequestScheme
+	  - ``String``
+	  -
+	* - owin.RequestMethod
+	  - ``String``
+	  -
+	* - owin.RequestPathBase
+	  - ``String``
+	  -
+	* - owin.RequestPath
+	  - ``String``
+	  -
+	* - owin.RequestQueryString
+	  - ``String``
+	  -
+	* - owin.RequestProtocol
+	  - ``String``
+	  -
+	* - owin.RequestHeaders
+	  - ``IDictionary<string,string[]>``
+	  -
+	* - owin.RequestBody
+	  - ``Stream``
+	  - 
+	 
+Request Data (OWIN v1.1.0)
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. list-table::
+	:header-rows: 1
+	
+	* - Key
+	  - Value (type)
+	  - Description
+	* - owin.RequestId
+	  - ``String``
+	  - Optional
+
+Response Data (OWIN v1.0.0)
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. list-table::
+	:header-rows: 1
+	
+	* - Key
+	  - Value (type)
+	  - Description
+	* - owin.ResponseStatusCode
+	  - ``int``
+	  - Optional
+	* - owin.ResponseReasonPhrase
+	  - ``String``
+	  - Optional
+	* - owin.ResponseProtocol
+	  - ``String``
+	  - Optional
+	* - owin.ResponseHeaders
+	  - ``IDictionary<string,string[]>``
+	  - 
+	* - owin.ResponseBody
+	  - ``Stream``
+	  -
+
+Other Data (OWIN v1.0.0)
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. list-table::
+	:header-rows: 1
+	
+	* - Key
+	  - Value (type)
+	  - Description
+	* - owin.CallCancelled
+	  - ``CancellationToken``
+	  -
+	* - owin.Version
+	  - ``String``
+	  -
+
+IAppBuilder.Properties
+^^^^^^^^^^^^^^^^^^^^^^
+
+.. list-table::
+	:header-rows: 1
+	
+	* - Key
+	  - Value (type)
+	  - Description
+	* - builder.AddSignatureConversion
+	  -
+	  -
+	* - builder.DefaultApp
+	  -
+	  -
+
+Common Keys
+^^^^^^^^^^^
+
+.. list-table::
+	:header-rows: 1
+	
+	* - Key
+	  - Value (type)
+	  - Description
+	* - ssl.ClientCertificate
+	  - ``X509Certificate``
+	  -
+	* - ssl.LoadClientCertAsync
+	  - ``Func<Task>``
+	  -
+	* - server.RemoteIpAddress
+	  - ``String``
+	  -
+	* - server.RemotePort
+	  - ``String``
+	  -
+	* - server.LocalIpAddress
+	  - ``String``
+	  -
+	* - server.LocalPort
+	  - ``String``
+	  -
+	* - server.IsLocal
+	  - ``bool``
+	  -
+	* - server.Capabilities
+	  - ``IDictionary<string,object>``
+	  -
+	* - server.OnSendingHeaders
+	  - ``Action<Action<object>,object>``
+	  -
+	* - host.TraceOutput
```

Comment:
host.TraceOutput is not supported. Check https://github.com/aspnet/HttpAbstractions/blob/dev/src/Microsoft.AspNet.Owin/OwinEnvironment.cs to verify what's supported.